zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/legacycacheapi/PayloadPathUpdater.java

```diff
@@ -0,0 +1,136 @@
+package com.github.onsdigital.zebedee.model.publishing.legacycacheapi;
+
+import com.github.onsdigital.zebedee.util.URIUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.github.onsdigital.logging.v2.event.SimpleEvent.error;
+
+public class PayloadPathUpdater {
+    public static final String BULLETIN_SEGMENT = "/bulletins";
+    // Segment input variables
+    private static final String ADHOC_SEGMENT = "/adhocs";
+    private static final String ARTICLE_SEGMENT = "/articles";
+    private static final String DATA_SEGMENT_WITHOUT_SLASH = "data";
+    private static final String DATASETS_SEGMENT = "/datasets";
+    private static final String LINE_CHART_CONFIG_SEGMENT_WITHOUT_SLASH = "linechartconfig";
+    private static final String QMIS_SEGMENT = "/qmis";
+    private static final String METHODOLOGY_SEGMENT = "/methodologies";
+    private static final String TIME_SERIES_SEGMENT = "/timeseries";
+    private static final String VISUALISATIONS_SEGMENT = "/visualisations";
+
+    // Resource types
+    private static final List<String> RESOURCE_TYPE_SEGMENTS = Arrays.asList(
+            "/chartconfig?",
+            "/chartimage?",
+            "/embed?",
+            "/chart?",
+            "/resource?",
+            "/generator?",
+            "/file?",
+            "/export?"
+    );
+
+    public static String getCanonicalPagePath(String incomingUrl, String collectionId) {
```

Comment:
This whole function seems very convoluted and prone to regression errors - if we had more time I would advise refactoring it.